### PR TITLE
media-sound/fluidsynth: bump to 2.0.4 with subslot, update ebuild

### PR DESCRIPTION
**Pull request by realnc**
Use subslot due to soversion change. Bump EAPI to 7. License is now LGPL-2.1+.
Add "oss" USE flag. Enable upstream-recommended USE flags by default (+readline
+sndfile). Drop versioned deps (no longer in portage.) Remove unused "ladcca"
cmake option. Put pkgconfig in BDEPEND. Remove forced "-pthread" flag.

FluidSynth 2.0 had some changes in its API that can potentially break some packages. I have tested the packages I use personally:

    games-engines/scummvm-2.0.0
    media-libs/sdl-mixer-1.2.12-r4
    media-libs/sdl2-mixer-2.0.4
    media-plugins/calf-0.90.0-r2
    media-sound/qsynth-0.5.3
    media-video/vlc-3.0.6-r1

They all build and work fine when rebuilding against fluidsynth-2.0.4 (preserved-rebuild will kick-in, since fluidsynth 2 changed the library from `so.1` to `so.2`.)

The next version of at least one existing package (games-engines/qtads), as well as one new package that will be submitted to portage soon, will require FluidSynth 2.x.

### CODE REVIEW

**Comment by a17r** [comment]
You want to check with `equery d --all fluidsynth` instead, should yield a bit more results. ;)

At the very least this version will have to be subslotted (`0/2` for soversion change).

**Comment by a17r** [comment]
> > At the very least this version will have to be subslotted (`0/2` for soversion change).
> I assume this means setting `SLOT="2"` in fluidsynth itself

No, that would be `SLOT="0/2"`. Same slot, because it is not something side-by-side installable with slot 0, but the subslot bump will ensure rebuild of dependencies that are (or will be) fixed to depend on `fluidsynth:=`

**Comment by realnc** [comment]
> No, that would be `SLOT="0/2"`. 

OK, done that.

> Same slot, because it is not something side-by-side installable with slot 0, but the subslot bump will ensure rebuild of dependencies that are (or will be) fixed to depend on `fluidsynth:=`

Should I be doing that for every package in a second commit in this PR, or is this up to each package's maintainer and thus something that will "just happen" over time?

File: media-sound/fluidsynth/fluidsynth-2.0.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	ladspa? (
+		media-libs/ladspa-cmt[${MULTILIB_USEDEP}]
+		media-libs/ladspa-sdk[${MULTILIB_USEDEP}]
+	)
+	portaudio? ( media-libs/portaudio[${MULTILIB_USEDEP}] )
+	pulseaudio? ( media-sound/pulseaudio[${MULTILIB_USEDEP}] )
+	readline? ( sys-libs/readline:0=[${MULTILIB_USEDEP}] )
+	sndfile? ( media-libs/libsndfile[${MULTILIB_USEDEP}] )"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig[${MULTILIB_USEDEP}]
```
Comment (a17r) [inline media-sound/fluidsynth/fluidsynth-2.0.4.ebuild]: this needs to go into `BDEPEND` instead for EAPI-7.

**Comment by a17r** [comment]
> Should I be doing that for every package in a second commit in this PR, or is this up to each package's maintainer and thus something that will "just happen" over time?

Checking/notifying can't hurt but is out of scope of this PR. I'll run a grep before merge for assessing the situation with barely- and unmaintained packages.

File: media-sound/fluidsynth/fluidsynth-2.0.4.ebuild (hunk 1 above)
Comment (realnc) [reply media-sound/fluidsynth/fluidsynth-2.0.4.ebuild]: Done. I also swapped RDEPEND and DEPEND, which seems more natural and is used in examples in the EAPI 7 documentation.

I also removed the forced `-pthread` flag. Bug 436762 is from 2012 and it seems it has been fixed long ago already in the build system.

File: media-sound/fluidsynth/fluidsynth-2.0.4.ebuild [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+LICENSE="LGPL-2.1+"
+SLOT="0/2"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ppc ~ppc64 ~sparc ~x86 ~x86-fbsd"
+IUSE="alsa dbus debug examples ipv6 jack ladspa lash oss portaudio pulseaudio +readline +sndfile"
+
+BDEPEND="
+	virtual/pkgconfig[${MULTILIB_USEDEP}]
+"
+DEPEND="
+	dev-libs/glib[${MULTILIB_USEDEP}]
```
[the remark this answers is not in the document]
Comment (realnc) [reply media-sound/fluidsynth/fluidsynth-2.0.4.ebuild]: Oops. Thanks for spotting that! Fixed.

**Pull request closed without merging**